Add `bordered` prop to `ComboBox` and update `ComboBox`'s read-only state (#3594)
jpmorganchase/salt-ds: .changeset/added-combobox-UI-fixes.md

---

> "@salt-ds/core": minor

Reviewer: ```suggestion
"@salt-ds/core": patch
```

---

> Updated background color prop of Validation status `PillInput` to reflect the color of the status.
> 
> Updated status adornment for Validation status `PillInput` to be visible for `readonly` state.

Reviewer: ```suggestion
Updated `ComboBox` to show validation status when it's in a read-only state.
```
